src/Compilers/Core/Portable/DiagnosticAnalyzer/AnalyzerExecutor.cs

```diff
@@ -803,15 +847,13 @@ private void ExecuteBlockActionsCore<TBlockStartAction, TBlockAction, TNodeActio
                     // Catch Exception from the start action.
                     ExecuteAndCatchIfThrows(
                         startAction.Analyzer,
-                        data =>
-                        {
-                            data.action(data.context);
-                            data.blockEndActions?.AddAll(data.scope.CodeBlockEndActions);
-                            data.syntaxNodeActions?.AddRange(data.scope.SyntaxNodeActions);
```

Comment:
these two lines don't need to be in the lambda .  we are protecting the call out to data.action as that is user code which may throw.  *after* taht, all we're doing is taking the data we collected and adding to the builders we own.  this is all code we own and cannot throw.  removing from teh lambda greatly simplifies the lambda (making it like all teh rest we have) and  greatly simplifes the data we pass to it.